internal/provider/resource_saml_idp.go

```diff
@@ -40,18 +45,18 @@ func resourceWizSAMLIdP() *schema.Resource {
 			},
 			"use_provider_managed_roles": {
 				Type:        schema.TypeBool,
-				Description: "Use provider managed roles?",
-				Optional:    true,
-				Default:     false,
+				Description: "Use provider (IdP) managed roles? This attribute needs to be set to accomodate for update lifecycle operations.",
+				Required:    true,
```

Comment:
Though the GraphQL schema labels this as required, this should remain optional with a default of false.  Default values cannot be defined for required attributes.

Reply:
The same comment as above, I was hoping there's a better approach here than to make all values default because the schema is not respected for patch ops once change gets introduced

Reply:
With this, we can achieve the desired effect with modification to the golang struct, such that empty values are not omitted.  The changes to the struct can be limited in scope to UpdateSAMLIdentityProviderPatch.

I ran into similar issues with other mutations.

Reply:
Ok, so I tracked down where I ran into a similar issue (different provider).  I used the [GetChange](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema#ResourceData.GetChange) function to get the old and new value, then handled the updates.

```
func accountUpdateString(key, cakey string, d *schema.ResourceData) interface{} {
	log.Printf("accountUpdate called for resource %s", d.Id())
	old, new := d.GetChange(key)
	_ = old
	if old == "" {
		update := &vendor.AccountUpdateString{
			Op:    "add",
			Path:  cakey,
			Value: new.(string),
		}
		return update
	} else if new == "" {
		update := &vendor.AccountUpdateRemove{
			Op:   "remove",
			Path: cakey,
		}
		return update
	} else {
		update := &vendor.AccountUpdateString{
			Op:    "replace",
			Path:  cakey,
			Value: new.(string),
		}
		return update
	}
}
```

We should do something similar to handle cases during update where a field was nullified or set to an empty value.  If we couple that with a deviation in struct UpdateSAMLIdentityProviderPatch, we can pass empty strings in the update variables, which should have the desired effect.